Support .NET Core 3.1

vebbo2 opened the pull request.

vebbo2 commented:
Hi @andrewlock , did you have a chance to take a look at this?

andrewlock commented:
Hi @vebbo2, sorry for the delay! Thanks for this - just wondering, was there any reason this wasn't working previously? As far as I'm aware, this should be working on .NET Core 3.1 shouldn't it?

andrewlock commented on test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj:
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
```
This should work if we're building for .NET Core 3.1 shouldn't it?

vebbo2 replied:
Yes, it should be fine

andrewlock replied:
I mean, we want our tests to run on .netcore 3, but it looks like that broke the build? Ideally, I'd like the tests to target multiple platforms to confirm it all works correctly 🙂

vebbo2 replied:
Oh, yes, AppVeyor seems to be configured to build with .NET Core 2.2. On my local machine it was obviously building correctly with .NET Core 3.1

andrewlock replied:
Looks like appveyor is missing the `image` tag for some reason! 

If you add it in (as in here):
https://github.com/andrewlock/PwnedPasswords/blob/2e3f117bcd3939cd53e25ab4e1ca15cea918398e/appveyor.yml#L2

Then we should be able to update the tests to multi-target all supported runtimes to be safe 🙂

```xml
<TargetFrameworks>netcoreapp2.1,netcoreapp3.0,netcoreapp3.1</TargetFrameworks>
```

Thanks!

vebbo2 replied:
Done, build succeeded

andrewlock replied:
Perfect, thanks!

vebbo2 commented:
@andrewlock Quoting you: https://github.com/andrewlock/StronglyTypedId/issues/5#issuecomment-524254501 :)
That's why CodeGeneration.Roslyn was upgraded and dotnet-codegen tool replaced with another package

andrewlock commented:
Haha, of course 🤦‍♂️ thanks for jogging my memory! 🙂

andrewlock commented:
Thanks @vebbo2, really appreciate you putting the effort in!😀

andrewlock commented:
I was just looking at this again @vebbo2 - I think we should probably update `StronglyTyped` to use the suggested metapackage configuration, as described in https://github.com/AArnott/CodeGeneration.Roslyn#create-the-metapackage?

vebbo2 commented:
@andrewlock I didn't exactly follow with that example because main `StronglyTypedId` already contains references to Attributes and Generator projects through dependencies in nuspec file. But when I removed these dependencies and moved it to csproj (like in the example), build was failing because nuget package cannot be build if project contains neither any nuspec dependency nor any source file :/ That's why I left it as it is

andrewlock commented:
@vebbo2 I've just made some tweaks in https://github.com/andrewlock/StronglyTypedId/pull/12 - Switched to using a 0-based enum for the Json type enum, and used the suggested metapackage config.

I'll push out a new release to NuGet shortly.

Thanks again for all your help! 😀
